src/character.cpp

```diff
@@ -484,26 +484,46 @@ VisitResponse Character::visit_items( const std::function<VisitResponse( item& )
     return inv.visit_items( func );
 }
 
-VisitResponse Character::visit_items( const std::function<VisitResponse( const item& )>& func ) const
+VisitResponse Character::visit_items( const std::function<VisitResponse( const item *, const item * )>& func ) const
```

Comment:
> 64 matches for "visit_items" on this page

Many of these are declarations as three classes implement their own version (plus a const overload) and then delegate to the next implementation. There's approximately a dozen actual calls to the function.

> Looks like you're overly specifying the visit function for that one usage.

`*::visit_items` is intended to provide the primitives (lookup, count, delete etc). As we add remainder of these eg. `*::remove_items_with` most will depend on this extra parameter

> adding the two argument function as an overload

This would work but I'm not sure if it's necessarily any cleaner as we now have four overloads of each function not two.

I agree however that the higher order functions should all use unary predicates eg. `std::function<bool(item&)>`.